feat: Add support for React Native Turbo Module

reyamir opened the pull request:
### Description
By support Turbo Module, React Native users can enjoy huge performance boost, it also allow using `nostr-sdk` with `hermes` enabled. See [Introducing Uniffi for React Native](https://hacks.mozilla.org/2024/12/introducing-uniffi-for-react-native-rust-powered-turbo-modules/)

### Notes to the reviewers

I've created a fork for `uniffi-bindgen-react-native` [here](https://github.com/reyamir/uniffi-bindgen-react-native), because it is using `uniffi-rs` version 0.29, but `rust-nostr` is using custom fork of `uniffi-rs`, it is causing version mismatch error, so I need to fork `uniffi-bindgen-react-native` and use custom fork of `uniffi-rs` like `rust-nostr`.

### Checklist

* [x] I followed the [contribution guidelines](https://github.com/rust-nostr/nostr/blob/master/CONTRIBUTING.md)
* [x] I ran `just precommit` or `just check` before committing
* [x] I updated the [CHANGELOG](https://github.com/rust-nostr/nostr/blob/master/CHANGELOG.md) (if applicable)

reyamir committed:
Init react native library

reyamir committed:
fix build

reyamir committed:
fix

reyamir committed:
add postinstall

reyamir committed:
remove postinstall script

yukibtc reviewed (changes requested):
Thanks! I'll check this better tomorrow. 

For now I have just a request: can you add also the script and/or the CI workflow to re/generate the bindings?

reyamir commented:
> Thanks! I'll check this better tomorrow.
> 
> For now I have just a request: can you add also the script and/or the CI workflow to re/generate the bindings?

I'll try

yukibtc commented:
FYI, I've bumped uniffi to 0.29.1 at https://github.com/rust-nostr/nostr-sdk-ffi/commit/aa5cebfad4075ca1a43a6490e9e274b96b0c4db9

reyamir committed:
add github ci

reyamir committed:
update ci

reyamir commented:
Hi @yukibtc, I've added GIthub CI actions based on this lib [https://github.com/unomed-dev/react-native-matrix-sdk](https://github.com/unomed-dev/react-native-matrix-sdk), the script seems fine, but look like I can only test it when it merged

yukibtc committed:
ci: fix branch name

yukibtc committed:
Fix nostr-sdk-ffi repo url and path in ubrn.config.yaml

yukibtc committed:
Bump uniffi-bindgen-react-native to 0.29.0-0

yukibtc committed:
Add nostr-sdk-ffi as git module

yukibtc committed:
Update package name and version

yukibtc committed:
Exclude `nostr-sdk-ffi` folder in tsconfig.build.json

yukibtc committed:
ci: consolidate workflows

yukibtc commented:
Thanks @reyamir! I've cleaned up the commits, fixed some stuff and merged everything in master branch. 

Currently there is only an issue with CI: it checks if during the building the files are changed. I regenerated all files but I can't regenerate the ones for the iOS, in the example, because I haven't a mac. Have you a mac? Are you able to regenerate them? I guess is enough to just run the example

yukibtc commented:
I set the `nostr-sdk-ffi` as git submodule, so after a `git pull`, you have to call also `git submodule update --init`

reyamir commented:
> Thanks @reyamir! I've cleaned up the commits, fixed some stuff and merged everything in master branch.
> 
> Currently there is only an issue with CI: it checks if during the building the files are changed. I regenerated all files but I can't regenerate the ones for the iOS, in the example, because I haven't a mac. Have you a mac? Are you able to regenerate them? I guess is enough to just rust the example

I've regenerate it fine on my mac

yukibtc commented:
> I've regenerate it fine on my mac

Nice! Are you able to commit the changes in a PR?

reyamir commented:
> > I've regenerate it fine on my mac
> 
> Nice! Are you able to commit the changes in a PR?

oh sorry, I've missed your reply, I will create a PR for iOS later
